eng/native/tryrun.cmake

```diff
@@ -170,8 +170,6 @@ else()
   message(FATAL_ERROR "Unsupported platform. OS: ${CMAKE_SYSTEM_NAME}, arch: ${TARGET_ARCH_NAME}")
 endif()
 
-if(TARGET_ARCH_NAME MATCHES "^(x86|x64|s390x|armv6|loongarch64|ppc64le)$")
+if(TARGET_ARCH_NAME MATCHES "^(x86|x64|s390x|armv6|loongarch64|riscv64|ppc64le)$")
   set_cache_value(HAVE_FUNCTIONAL_PTHREAD_ROBUST_MUTEXES_EXITCODE 0)
```

Comment:
@kouvel Do we still need `HAVE_FUNCTIONAL_PTHREAD_ROBUST_MUTEXES`? It does not look right that it is architecture specific.

Reply:
It was introduced due to a couple of issues, one in Alpine Linux and one in glibc on ARM/ARM64, but I believe that both have since been fixed. I'm not sure why the cache value is being set based on architecture, it should probably be checked based on the target OS, which is currently done in `mutex.hpp` in the PAL.

It was noted in https://github.com/dotnet/runtime/pull/82380#discussion_r1128826231 that a PAL test for named mutexes fails when pthread robust mutexes are used on RISC-V, so I wonder if it actually works. @JongHeonChoi, have you checked if the corresponding test case in https://github.com/dotnet/runtime/blob/e235aeff6c822ad18a63b39819f61031a1ac95e2/src/coreclr/pal/src/configure.cmake#L1012-L1199 compiles and runs successfully on RISC-V? Probably would also need to check the PAL named mutex tests to make sure that it works.

Reply:
@kouvel Yes. I checked the return value of the test case in VisionFive2 target.

Reply:
Sounds good, thanks! I'll look into whether this check can be removed, as I'm not aware of any current cases where it would fail.